File: collatz/collatz.jl
function typed_flat_collantz(b::Int64)::Tuple{Int64,Int64}
    ms::Tuple{Int64,Int64} = (0,0)
    while b > 0
        n::Int64 = b
        c::Int64 = 0
        while n > 1
            c += 1
            n = if n % 2 == 1  3 * n + 1 else n / 2 end
        end
        if c > ms[2] ms = (b,c) end
        b -= 1
    end
    ms
end
print(max_collantz_speed(1,100000000))

File: collatz/collatz.py
def max_collantz_speed(x:int,n:int)->tuple:
    ms = (0,0)
    next = lambda x: 3 * x + 1 if x % 2 else x / 2 
    speed = lambda s: 1 + speed(next(s)) if s - 1 else 0      
    while (x:=x+1) < n+1:
        if (s:= speed(x)) > ms[1]:
            ms = (x,s)
    return ms

print(max_collantz_speed(1,10_000_000))
